extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java

```diff
@@ -490,21 +496,23 @@ private WebSocketEndpointBuildItem findEndpoint(String endpointId, List<WebSocke
         throw new IllegalArgumentException("Endpoint not found: " + endpointId);
     }
 
-    private boolean needsRequestContext(BeanInfo bean, Set<String> processedBeans, BeanResolver beanResolver) {
+    private boolean needsContext(BeanInfo bean, ScopeInfo scope, Set<String> processedBeans, BeanResolver beanResolver) {
         if (processedBeans.add(bean.getIdentifier())) {
-            if (BuiltinScope.REQUEST.is(bean.getScope())
-                    || (bean.isClassBean()
-                            && bean.hasAroundInvokeInterceptors()
-                            && SecurityTransformerUtils.hasSecurityAnnotation(bean.getTarget().get().asClass()))) {
-                // Bean is:
-                // 1. Request scoped, or
-                // 2. Is class-based, has an aroundInvoke interceptor associated and is annotated with a security annotation
+
+            if (scope.equals(bean.getScope())) {
+                // Bean has the given scope
+                return true;
+            } else if (BuiltinScope.REQUEST.is(scope)
+                    && bean.isClassBean()
+                    && bean.hasAroundInvokeInterceptors()
+                    && SecurityTransformerUtils.hasSecurityAnnotation(bean.getTarget().get().asClass())) {
+                // The given scope is RequestScoped, the bean is class-based, has an aroundInvoke interceptor associated and is annotated with a security annotation
```

Comment:
CI is green, so it is probably unnecessary question, but this bean info comes from the `ValidationPhaseBuildItem` and I think this item already reflects `AnnotationsTransformerBuildItem`, so I am surprised that endpoints would have associated CDI interceptors. I thought they are removed. I suppose that class info does not reflect changes and you would need to use annotation store, but how is it that `hasAroundInvokeInterceptors` is true?

Reply:
That's actually a very good question. I completely forgot that we remove the security annotations from endpoints. However, this PR does not change the logic, i.e. this condition was introduced in https://github.com/quarkusio/quarkus/pull/43915: https://github.com/quarkusio/quarkus/pull/43915/files#diff-ceb74c4c03f53a1843b0c3b46d77068d483c1694677307b1415a86d6d91a8a34R487-R495.

Hm, so the annotations are removed and security interceptors should not be associated. But I guess that it's fine because we don't use the injected `CurrentIdentityAssociation` (which is `@RequestScoped`) for endpoint validation.

Reply:
Right, that is true, we don't need `CurrentIdentityAssociation` for endpoint security checks. I didn't realize that. Cool, in that case it is fine.